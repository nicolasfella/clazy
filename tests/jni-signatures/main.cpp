
// Crude approximation of #include <QAndroidJniObject> so we don't need QtAndroidExtras installed
class QAndroidJniObject
{
    public:
    QAndroidJniObject(const char *className, const char *signature);

    template <typename T>
    void callObjectMethod(const char *methodName, const char *signature);

    template <typename T>
    void callMethod(const char *methodName, const char *signature);

    template <typename T>
    static void callStaticMethod(const char *methodName, const char *signature);

    template <typename T>
    static void callStaticObjectMethod(const char *methodName, const char *signature);
};

int main() {

    QAndroidJniObject obj("java/lang/String", "(");
    obj.callObjectMethod<int>("someMethod", "()V");
    obj.callObjectMethod<int>("someMethod", "(III)V");
    obj.callObjectMethod<int>("someMethod", "(III");

    obj.callMethod<int>("someMethod", "()V");
    obj.callMethod<int>("someMethod", "(III)V");
    obj.callMethod<int>("someMethod", "()Ljava/lang/String");

    QAndroidJniObject::callStaticMethod<int>("someMethod", "()Ljava/lang/String");

    QAndroidJniObject::callStaticObjectMethod<int>("someOtherMethod", "(II)");
}
