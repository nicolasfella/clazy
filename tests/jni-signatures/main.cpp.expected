jni-signatures/main.cpp:23:23: warning: Invalid method signature: '(' [-Wclazy-jni-signatures]
jni-signatures/main.cpp:26:5: warning: Invalid method signature: '(III' [-Wclazy-jni-signatures]
jni-signatures/main.cpp:30:5: warning: Invalid method signature: '()Ljava/lang/String' [-Wclazy-jni-signatures]
jni-signatures/main.cpp:32:5: warning: Invalid method signature: '()Ljava/lang/String' [-Wclazy-jni-signatures]
jni-signatures/main.cpp:34:5: warning: Invalid method signature: '(II)' [-Wclazy-jni-signatures]
